test/headers.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {Storage} from '../src';
+import * as nock from 'nock';
+
+describe('headers', () => {
+  before(() => {
+    nock.disableNetConnect();
+  });
+  it('populates x-goog-api-client header', async () => {
+    const storage = new Storage({
+      projectId: 'foo',
+    });
+    const bucket = storage.bucket('foo-bucket');
+    const metadata = nock('http://metadata.google.internal.')
```

Comment:
This is going to be a problem.  On my machine, I am logged in with the `Cloud SDK`.  I also have a `GOOGLE_APPLICATION_CREDENTIALS` var set.  With existing logins created ... the auth library will never get to the metadata service to try talking to it.  

This test would fail due to the `disableNetconnect` above, and I'd be super confused why my unit tests were failing.

Instead - I think we should use proxyquire, and stub out the `getClient` call, or the `request` call, or something like that.

I'm open to other ideas, but here's what I've done:
https://github.com/JustinBeckwith/gcbuild/blob/master/test/test.ts#L18

Reply:
I see no reason to not just use `proxyquire`.